Graphics/GraphicsEngineD3D11/include/PipelineStateD3D11Impl.hpp

```diff
@@ -142,9 +142,9 @@ class PipelineStateD3D11Impl final : public PipelineStateBase<IPipelineStateD3D1
     // SRB memory allocator must be defined before the default shader res binding
     SRBMemoryAllocator m_SRBMemAllocator;
 
-    Int8 m_ResourceLayoutIndex[6] = {-1, -1, -1, -1, -1, -1};
+    std::array<Int8, NUM_SHADER_TYPES> m_ResourceLayoutIndex;
```

Comment:
We can have at most 5 different shaders at in one pipeline. Now there are 3 orthogonal pipeline types. With ray tracing, there will be another pipeline type and 5 more shader stages.
A proper solution is to add pipeline type parameter to GetShaderTypeIndex function to make indices overlap.
I will work on this.

Reply:
Pipeline type is not needed in GetShaderTypeIndex:
```cpp
case SHADER_TYPE_PIXEL: return 0;
case SHADER_TYPE_VERTEX:
case SHADER_TYPE_AMPLIFICATION: return 1;
case SHADER_TYPE_GEOMETRY:
case SHADER_TYPE_MESH: return 2;
...
```

Reply:
Yes, I was thinking about that too. I may add the pipeline type for debug purposes, just to make sure that e.g. mesh shader is not requested for compute pipeline.